Kudu.Core/Infrastructure/PathUtility.cs

```diff
@@ -136,10 +136,23 @@ private static string ResolveNpmVersion()
         private static string ResolveNpmVersion(string nodeVersion)
         {
             string programFiles = SystemEnvironment.GetFolderPath(SystemEnvironment.SpecialFolder.ProgramFilesX86);
+            string appSettingNpmVersion = SystemEnvironment.GetEnvironmentVariable("APPSETTING_WEBSITE_NPM_DEFAULT_VERSION");
```

Comment:
Just `WEBSITE_NPM_DEFAULT_VERSION` should work too and is cleaner for the non-Azure case.